patches/minecraft/net/minecraft/world/level/block/FlowerBlock.java.patch

```diff
@@ -0,0 +1,41 @@
+--- a/net/minecraft/world/level/block/FlowerBlock.java
++++ b/net/minecraft/world/level/block/FlowerBlock.java
+@@ -15,6 +_,17 @@
+    private final MobEffect f_53508_;
+    private final int f_53509_;
+ 
++   private final java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier;
++
++   public FlowerBlock(java.util.function.Supplier<MobEffect> suspiciousStewEffectSupplier, int p_53513_, BlockBehaviour.Properties p_53514_) {
++      super(p_53514_);
++      this.f_53508_ = null;
++      this.suspiciousStewEffectSupplier = suspiciousStewEffectSupplier;
++      this.f_53509_ = p_53513_;
++   }
++
++   /** @deprecated FORGE: Use supplier version instead */
++   @Deprecated
+    public FlowerBlock(MobEffect p_53512_, int p_53513_, BlockBehaviour.Properties p_53514_) {
+       super(p_53514_);
+       this.f_53508_ = p_53512_;
+@@ -23,6 +_,7 @@
+       } else {
+          this.f_53509_ = p_53513_ * 20;
+       }
++      this.suspiciousStewEffectSupplier = null;
```

Comment:
This should either return the mob effect supplied in the constructor or, to support registry replacement, a delegate similar to `LiquidBlock` or just by calling `RegistryObject#of`.